test/NuGet.Core.Tests/NuGet.Packaging.Test/SigningTests/SignedPackageVerifierTests.cs

```diff
@@ -1,127 +1,55 @@
-        {
-            var signature = new Signature()
-            {
-                DisplayName = "Test Signer",
-                TestTrust = SignatureVerificationStatus.Invalid,
-                Type = SignatureType.Author
-            };
-
-            var verifyResult = await GetTrustResultAsync(signature, SignedPackageVerifierSettings.AllowAll);
-
-            verifyResult.Valid.Should().BeFalse();
-            verifyResult.Results.Single().Trust.Should().Be(SignatureVerificationStatus.Invalid);
-        }
-
-        // Verify a package with no signature
+        // Verify a valid and trusted signature on a package.
         [Fact]
-        public async Task SignedPackageVerifier_CreateUnSignedPackageVerifySignatureWithAllowAll()
-        {
-            var verifyResult = await GetTrustResultAsync(null, SignedPackageVerifierSettings.AllowAll);
-
-            verifyResult.Valid.Should().BeTrue();
-            verifyResult.Results.Should().BeEmpty();
-        }
-
-        // Verify a package with no signature and require signing
-        [Fact]
-        public async Task SignedPackageVerifier_CreateUnSignedPackageVerifySignatureWithRequireSigned()
-        {
-            var verifyResult = await GetTrustResultAsync(null, SignedPackageVerifierSettings.RequireSigned);
-
-            verifyResult.Valid.Should().BeFalse();
-            verifyResult.Results.Should().BeEmpty();
-        }
-
-        private static Task<VerifySignaturesResult> GetTrustResultAsync(Signature signature, SignedPackageVerifierSettings settings)
+        public async Task SignedPackageVerifier_CreateValidPackageVerifyTrustedResult()
         {
             var nupkg = new SimpleTestPackageContext();
-            if (signature != null)
-            {
-                nupkg.Signatures.Add(signature);
-            }
-
             var testLogger = new TestLogger();
             var zip = nupkg.Create();
 
+            using (var testCert = TestCertificate.Generate().WithTrust())
             using (var signPackage = new SignedPackageArchive(zip))
             {
-                var trustProviders = new[] { new SignatureVerificationProvider() };
-                var verifier = new SignedPackageVerifier(trustProviders, settings);
+                await SignTestUtility.SignPackageAsync(testLogger, testCert.Source.Cert, signPackage);
 
-                return verifier.VerifySignaturesAsync(signPackage, testLogger, CancellationToken.None);
+                var settings = SignedPackageVerifierSettings.RequireSigned;
+
+                var result = await VerifySignatureAsync(testLogger, signPackage, settings);
+
+                result.Valid.Should().BeTrue();
             }
         }
+
+        private static async Task<VerifySignaturesResult> VerifySignatureAsync(TestLogger testLogger, SignedPackageArchive signPackage, SignedPackageVerifierSettings settings)
+        {
+            var trustProviders = new[] { new X509SignatureVerificationProvider() };
```

Comment:
nit: Change `trustProviders` to `verificationProviders`

Reply:
fixed